### PR TITLE
fix(ZNTA-2395): references to http

### PR DESCRIPTION
**Pull request by kathryngo**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2395

Fix errors such as the following in react editor

```
Mixed Content: The page at 'https://translate.stage.engineering.redhat.com/project/translate/odys/v/1/odyssey.txt?dswid=9827&lang=de'c was loaded over HTTPS, but requested an insecure image 'http://www.gravatar.com/avatar/undefined?d=mm&ampr=g&amps=72'. This content should also be served over HTTPS.
```

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/721)

### CODE REVIEW

File: server/zanata-frontend/src/app/editor/index.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
   <link rel="icon" type="image/x-icon" href="http://zanata.org/images/logo/logo.ico">
   <link rel="stylesheet" type="text/css" href="/editor.css">
   <link rel="stylesheet" type="text/css" async
-    href='http://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,600,700,400italic'>
+    href='https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,600,700,400italic'>
```
Comment (seanf) [inline server/zanata-frontend/src/app/editor/index.html]: How did you decide whether to change `http` -> `https` or to remove the protocol entirely? Most of the other changes omit the protocol.

Actually, reading a bit more about it (eg https://jeremywagner.me/blog/stop-using-the-protocol-relative-url/) I think we should unconditionally use `https`, both for security and speed reasons (eg HTTP2). 

Sorry to change things on you.

File: server/zanata-frontend/src/app/editor/components/DashboardLink/DashboardLink.test.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import DashboardLink from '.'
 describe('DashboardLinkTest', () => {
   it('DashboardLink markup', () => {
     const profilePicSrc =
-      'http://www.emoji-cheat-sheet.com/graphics/emojis/smiling_imp.png'
+      '//www.emoji-cheat-sheet.com/graphics/emojis/smiling_imp.png'
```
Comment (seanf) [inline server/zanata-frontend/src/app/editor/components/DashboardLink/DashboardLink.test.js]: I know this is just test code, but it seems the URL is no longer pointing at an image (especially when you put `https` in front of it). This works now: https://www.webpagefx.com/tools/emoji-cheat-sheet/graphics/emojis/smiling_imp.png

File: server/zanata-frontend/src/app/editor/index.html (hunk 1 above)
Comment (kathryngo) [reply server/zanata-frontend/src/app/editor/index.html]: No problem. Thought we should be using https where possible.

File: server/zanata-frontend/src/app/editor/components/DashboardLink/DashboardLink.test.js (hunk 2 above)
Comment (kathryngo) [reply server/zanata-frontend/src/app/editor/components/DashboardLink/DashboardLink.test.js]: Ok thanks, will update the code.

File: server/zanata-frontend/src/app/editor/index.html (hunk 1 above)
Comment (seanf) [reply server/zanata-frontend/src/app/editor/index.html]: What I mean is, could you please change all five of the `//www.gravatar.com` links to `https//www.gravatar.com` too? There's four of them in this PR, but there's one in src/index.html too.

Comment (kathryngo) [reply server/zanata-frontend/src/app/editor/index.html]: No problem. Sorry for the misunderstanding

**Comment by kathryngo** [comment]
Review status: 0 of 8 files reviewed at latest revision, 5 unresolved discussions.

---

*[server/zanata-frontend/src/app/editor/reducers/header-data-reducer.js, line 47 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/721#-L5CLnRD91Q45fMw62Bl-r4-47:-L5RdexZinAV1-KrsQcx:b-tj5hk4) ([raw file](https://github.com/zanata/zanata-platform/blob/00612e79d6af4c2275f663328d67bdb671b9860e/server/zanata-frontend/src/app/editor/reducers/header-data-reducer.js#L47)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Again, this was already broken (I think), but I think we need to replace `&amp` with `&` in this URL, because `ampr` and `amps` don't seem to be valid parameters, but `r` and `s` are (rating and size). https://en.gravatar.com/site/implement/images/

Unless this string goes through some sort of decoding before we use it. Do you think so?
</blockquote></details>

Aha I think you are right about the r & s parameters. I don't think there is any fancy decoding going on here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/721)*

File: server/zanata-frontend/src/app/editor/reducers/header-data-reducer.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ const defaultState = {
 }
 
 const gravatarUrl = (hash, size) => {
-  return `https://www.gravatar.com/avatar/${hash}?d=mm&ampr=g&amps=${size}`
+  return `https://www.gravatar.com/avatar/${hash}?d=mm&=g&s=${size}`
```
Comment (seanf) [inline server/zanata-frontend/src/app/editor/reducers/header-data-reducer.js]: missing the `r`

File: server/zanata-frontend/src/app/editor/reducers/header-data-reducer.test.js [hunk 4]
```diff
@@ -99,7 +99,7 @@ describe('header-data-reducer test', () => {
     expect(withData).toEqual({
       user: {
         name: 'rick',
-        gravatarUrl: 'https://www.gravatar.com/avatar/12345?d=mm&ampr=g&amps=72',
+        gravatarUrl: 'https://www.gravatar.com/avatar/12345?d=mm&=g&s=72',
```
Comment (seanf) [inline server/zanata-frontend/src/app/editor/reducers/header-data-reducer.test.js]: here too.

**Comment by djansen-redhat** [comment]
Tester notes: gravatar profile image still works, no 'Mixed content' error in console

**Comment by seanf** [comment]
@djansen-redhat Note that you would have to run the app server via https (or proxy it via https) to see the mixed content warning.

**Comment by seanf** [comment]
@djansen-redhat However, if these https images work in http mode, they should keep working when deployed behind an https proxy, so if the images are working that's the main thing.

**Comment by seanf** [comment]
:lgtm:

---

Reviewed 1 of 6 files at r1, 1 of 1 files at r2, 3 of 6 files at r4, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/721#-:-L5S7baGcSSj6BG-UL_k:bnfp4nl)*

**Comment by kathryngo** [comment]
@djansen-redhat has this got the QA seal of approval?

**Comment by spathare21** [comment]
@djansen-redhat are you looking into it? I am not sure how to test this one locally

**Pull request merged**